### ISSUE TITLE
Consistent crash - Tried to get a plugin before it was configured

### ISSUE THREAD

**Issue by bpappin** [aws-amplify/amplify-android#1003]
We get this crash on all kind of devices, build after build.

```
2020-11-26 22:15:05.738 11052-11052/ambimi.rogue.dev E/AndroidRuntime: FATAL EXCEPTION: main
    Process: ambimi.rogue.dev, PID: 11052
    java.lang.IllegalStateException: Tried to get a plugin before it was configured.  Make sure you call Amplify.configure() first.
        at com.amplifyframework.core.category.Category.getSelectedPlugin(Category.java:225)
        at com.amplifyframework.storage.StorageCategory.uploadInputStream(StorageCategory.java:170)
...
```

Amplify is being initialized in the Application class as:

```
      try {
            // Add these lines to add the AWSCognitoAuthPlugin and AWSS3StoragePlugin plugins
            Amplify.addPlugin(AWSCognitoAuthPlugin())
            Amplify.addPlugin(AWSS3StoragePlugin())
            Amplify.configure(applicationContext)
            Log.i(TAG, "Initialized Amplify")
        } catch (error: AmplifyException) {
            Log.e(TAG, "Could not initialize Amplify", error)
        }
```

The configuration happens in the application _onCreate_ method, and the call to upload happens long after the application is running, yet we consistently get this during testing.

**Comment by jamesonwilliams** [comment]
@bpappin is this the same as #1002? Please include a broader context of logs from your application startup, including verbose output. See the "Getting More Output" section here, to enable it: https://github.com/aws-amplify/amplify-android/blob/main/CONTRIBUTING.md

**Comment by bpappin** [comment]
I saw that one. Although the error looks similar, it doesn't appear to be the same issue.

We are getting a slew of problems with Amplify, the latest one seems to be its looking for a config file, that the tool added to the .gitignore list.

**Comment by jamesonwilliams** [comment]
Ah, I get it. This #1003 will definitely happen if #1005 happened first. Resolving your usage in #1005 will fix this current #1003.
